### PR TITLE
HARP-10648: Fix stroking of geometry touching the tile border.

### PR DESCRIPTION
**Pull request by robertoraggi**
This change simplifies the support for working with rings
and texture coordinates. The Ring now maintains distinct
arrays of position and texture coordinates and has APIs
to flatten the Ring into an Array of numbers.

### CODE REVIEW

File: test/rendering/GeoJsonDataRendering.ts [hunk 1]
```diff
@@ -287,4 +288,131 @@ describe("MapView + OmvDataSource + GeoJsonDataProvider rendering test", functio
...
+            theme: { lights, styles: { geojson: ourStyle } },
+            geoJson,
+            lookAt: {
+                target: [0, 51.5],
+                zoomLevel: 13.4
+            }
+        });
+    });
+
+    it("renders geometries touching and crosssing tile bounds", async function() {
```
Comment (nzjony) [inline test/rendering/GeoJsonDataRendering.ts:341]:
```suggestion
    it("renders geometries touching and crossing tile bounds", async function() {
```

File: @here/harp-omv-datasource/lib/Ring.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            );
+        }
+
+        this.vertexStride = 2;
+
+        if (textureCoords !== undefined) {
+            this.vertexStride = this.vertexStride + 2;
+        }
+
+        this.area = ShapeUtils.area(this.points);
```
[the remark this answers is not in the document]
Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/Ring.ts:76]: I changed the documentation to explicit mention that we store the signed area of the ring.

File: @here/harp-omv-datasource/lib/Ring.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * Tests if the edge connecting the vertex at `index` with
+     * the vertex at `index+1` should be connected by an outline
+     * when stroking the polygon.
+     *
+     * @param index The index of the first vertex of the outline edge.
+     */
+    isOutline(index: number): boolean {
```
[the remark this answers is not in the document]
Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/Ring.ts]: I renamed it to `edgeIsSynthetic`

[the remark this answers is not in the document]
Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/Ring.ts]: I renamed to `isProperEdge`

File: test/rendering/GeoJsonDataRendering.ts (hunk 1 above)
Comment (robertoraggi) [reply test/rendering/GeoJsonDataRendering.ts]: fixed

**Pull request merged**